backend/src/controllers/filter-item-controller.ts

```diff
@@ -0,0 +1,166 @@
+import { Request, Response } from 'express'; 
+import { validationResult } from 'express-validator'; 
+import InventoryItem from '../models/inventory-item'; 
+import Item from '../models/item';
+import mongoose from 'mongoose';
+
+export const getFilteredInventoryItems = async (req: Request, res: Response) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        console.log("errors", errors);
+        return res.status(422).json({ errors: errors.array() });
+    }
+
+    try {
+        // Filter
+        let fetchedInventoryItems = {};
+        const itemOwner = req.query.itemOwner as string
+        const size = req.query.size as string
+        const name = req.query.name as string
+
+        // Sort
+
+        const sortedBy = req.query.sortedBy as string
+
+        if (itemOwner && size && name) {
+            let tempData: any[] = await Item.find({})
+            let tempDataObject: any;
+            if (!tempData) {
+                res.status(404).json({ message: "Not Found" })
+            }
+
+            tempData.forEach((data) => {
+                const isIdValid = mongoose.Types.ObjectId.isValid(data._id) 
+    
+                if (!isIdValid) {
+                    return res.status(500).json({ message: "Invalid Data" }) 
+                }
+                
+                if (data.name.toLowerCase().includes(name.toLowerCase()) && data.itemOwner.toLowerCase().includes(itemOwner.toLowerCase())) {
+                    tempDataObject = data._id
+                }  
+            })
+            fetchedInventoryItems = {itemId: tempDataObject, size: size, amount: {$gt: 1}}
```

Comment:
Calling to the route should return all items with child inventoryItems that meet the above requirement, not just one that it finds

Reply:
Issue repeats itself in the other loops too